### PR TITLE
Sd 576 implement payoutservice methods

### PR DESCRIPTION
**Pull request by filipdujmusic**

### CODE REVIEW

**Comment by filipdujmusic** [comment]
@githinho @domagojlatecki 

- moved payout-manager tests to test/managers package
- added tests for payout-service
- extracted merkleTreeValidator contract deployment to separate function and env var so we can reuse it if merkleTreeValidator remains the same but payoutManager is updated (as suggeseted by LaT)

I have to merge this one to the  sd-577-replace-deprecated-snapshotdistributor before merging everything to master.

File: test/services/payout-service-test.ts [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+                testData.assetFactory.address,
+                testData.assetTransferableFactory.address,
+                testData.assetSimpleFactory.address
+            ]
+        );
+
+        expect(payoutsAfterCreation.length).to.be.equal(3);
+        
+        const assetBasicPayout: PayoutForIssuerItem = 
+            payoutsAfterCreation.filter(item => item.asset == assetBasic.address)[0];
```
Comment (githinho) [inline test/services/payout-service-test.ts:157]: Just info, you could use find for this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

File: test/services/payout-service-test.ts [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        expect(fetchedPayoutForIssuerItem.assetSnapshotBlockNumber).to.be.equal(expectedPayoutInfo.assetSnapshotBlockNumber);
+        expect(fetchedPayoutForIssuerItem.assetSnapshotMerkleIpfsHash).to.be.equal(expectedPayoutInfo.assetSnapshotMerkleIpfsHash);
+        expect(fetchedPayoutForIssuerItem.rewardAsset).to.be.equal(expectedPayoutInfo.rewardAsset);
+        expect(fetchedPayoutForIssuerItem.totalRewardAmount).to.be.equal(expectedPayoutInfo.totalRewardAmount);
+        expect(fetchedPayoutForIssuerItem.remainingRewardAmount).to.be.equal(expectedPayoutInfo.totalRewardAmount);
+    }
+
+    async function createPayouts() {
+
+    }
```
Comment (githinho) [inline test/services/payout-service-test.ts:259]: Take no prisoners, use or delete.

File: test/services/payout-service-test.ts [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        payoutInfo: string,
+        assetSnapshotMerkleRoot: string,
+        assetSnapshotMerkleDepth: BigNumber,
+        assetSnapshotBlockNumber: BigNumber,
+        assetSnapshotMerkleIpfsHash: string,
+        totalRewardAmount: BigNumber,
+        rewardAsset: string
+    };
+
+    interface PayoutForIssuerItem {
```
Comment (githinho) [inline test/services/payout-service-test.ts:217]: Is it possible to use generated interface from `Payout` defined in `Structs` contract?

Comment (filipdujmusic) [reply test/services/payout-service-test.ts:217]: seems it's not possible to do this , did a bit of googling.
For the rest of the comments I've fixed them

**Review by githinho — APPROVED** [review]

File: test/services/payout-service-test.ts (hunk 3 above)
Comment (domagojlatecki) [reply test/services/payout-service-test.ts:217]: Yeah, it's very annoying - typechain generated values return something that is both array and an object, e.g. `[string, number] & { something: string, other: number }`. (Java|Type)Script black magic I guess 🤷‍♂️

**Review by domagojlatecki — APPROVED** [review]

**Pull request merged**